### PR TITLE
scripts: easy tool to quick setup the cluster from zero

### PR DESCRIPTION
**Pull request by leyao-daily**
This PR introduces an orchestrated procedure to setup a host machine and initialize or join a Kubernetes cluster. The script is designed to ensure host readiness, disable swap, configure kernel parameters, install necessary prerequisites like Docker and Kubernetes packages, setup Containerd, initialize or join the cluster, and optionally setup a Container Network Interface (CNI) plugin.

### CODE REVIEW

**Comment by jiere** [comment]
May I know why use Python to organize such scripts? Since we have "tools" directory also in this repo which hosts shell scripts there.

**Comment by leyao-daily** [comment]
> May I know why use Python to organize such scripts? Since we have "tools" directory also in this repo which hosts shell scripts there.

Tools for the cnap related shell scripts, and this is used to setup k8s quick on a `new setup machine`, so i just move it to k8s-mainifests, one can execute it to setup or join cluster quick, especially for some testing or demo.

Using python can brings more feature and flexiblities to it. Maybe cross platform bulabula..

Of course, if we don't need this to setup k8s, i can move it to other repo or location. :)

**Comment by jiere** [comment]
> Using python can brings more feature and flexiblities to it. Maybe cross platform bulabula..
> 
> Of course, if we don't need this to setup k8s, i can move it to other repo or location. :)

Yeah, what I am thinking is just about where to put such deployment related scripts, since our current demo repo is not really open sourced in public, but for real open source project, deployment related scripts are usually in Shell/Ansible/helm(yaml), etc.

**Comment by leyao-daily** [comment]
> > Using python can brings more feature and flexiblities to it. Maybe cross platform bulabula..
> > Of course, if we don't need this to setup k8s, i can move it to other repo or location. :)
> 
> Yeah, what I am thinking is just about where to put such deployment related scripts, since our current demo repo is not really open sourced in public, but for real open source project, deployment related scripts are usually in Shell/Ansible/helm(yaml), etc.

what do yout think a directory`utilities` or `dependencies`, because k8s is indeed the dependency of the project?

File: README.md [hunk 1]
```diff
@@ -53,6 +53,12 @@ The Dockerfile is under the directories in [container](container/)
 
 ## 4. Deployment
 
+Before you deploy the helm chart, you need to setup the kubernetes cluster and install the helm tool. Please refer to the [kubernetes documentation](https://kubernetes.io/docs/setup/) and [helm documentation](https://helm.sh/docs/intro/install/) for more details. We also deliver a quick start script to setup the kubernetes cluster:
+
+```bash
+python3 ./tools/prerequisites/k8s-setup.py
```
Comment (Hulongyin) [inline README.md]: Can add a note to indicate that it's for Ubuntu

Comment (leyao-daily) [reply README.md]: fixed. change the script with ubuntu tag

**Review by jiere — APPROVED** [review]
Current directory LGTM.
Since we have involved KEDA/Kepler in this project scope in README.md, I will raise PR to include the Kepler deployment later on top of your work:)
How about KEDA then? Maybe KEDA/Kepler scripts could be placed under tools/integration?

**Comment by leyao-daily** [comment]
> Current directory LGTM. Since we have involved KEDA/Kepler in this project scope in README.md, I will raise PR to include the Kepler deployment later on top of your work:) How about KEDA then? Maybe KEDA/Kepler scripts could be placed under tools/integration?

We have keda scable job for cnap yet, and for keda installation, we deliver the document under ./docs

**Review by Hulongyin — APPROVED** [review]
LGTM. Thanks

**Pull request merged**